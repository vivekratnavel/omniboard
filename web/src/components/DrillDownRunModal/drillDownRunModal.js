import PropTypes from 'prop-types';
import React, {PureComponent} from 'react';
import {Modal, ModalHeader, ModalBody, ModalTitle} from 'react-bootstrap';
import './drillDownRunModal.scss';
import {DrillDownView} from '../DrillDownView/drillDownView';

class DrillDownRunModal extends PureComponent {
  static propTypes = {
    handleClose: PropTypes.func.isRequired,
    shouldShow: PropTypes.bool.isRequired,
    runId: PropTypes.number,
    status: PropTypes.string,
    localStorageKey: PropTypes.string,
    dbInfo: PropTypes.shape({
      path: PropTypes.string,
      key: PropTypes.string
    }).isRequired
  };

  render() {
    const {shouldShow, handleClose, runId, dbInfo, status, localStorageKey} = this.props;
    if (runId === null) {
      return (
        <Modal dialogClassName='drilldown-run-modal' show={shouldShow} onHide={handleClose}>
          <ModalHeader closeButton>
            <ModalTitle>Error</ModalTitle>
          </ModalHeader>
          <ModalBody>
            <div><strong>Error: Run Id cannot be null.</strong></div>
          </ModalBody>
        </Modal>
      );
    }

    const modalTitle = `Details for Run Id: ${runId}`;
    const drillDownView = (
      <DrillDownView dbInfo={dbInfo}
        runId={runId}
        status={status}
        localStorageKey={localStorageKey}
        width='100%'
        height='100%'
      />
    );

    return (
      <Modal dialogClassName='drilldown-run-modal' show={shouldShow} onHide={handleClose}>
        <ModalHeader closeButton>
          <ModalTitle>{modalTitle}</ModalTitle>
        </ModalHeader>
        <ModalBody>
          {drillDownView}
        </ModalBody>
      </Modal>
    );
  }
}

export {DrillDownRunModal};
