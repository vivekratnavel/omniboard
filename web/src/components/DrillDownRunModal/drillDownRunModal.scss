$SIDEBAR_WIDTH: 200;
$HEADER_HEIGHT: 52;
$light-steel-blue: #b0c4de;

.full-height {
  height: 100%;
}

.drilldown-run-modal {
  width: 95vw;
  height: calc(95vh - 15px);
  .modal-content {
    height: 100%;
  }
  .modal-header {
    height: #{$HEADER_HEIGHT}px;
  }
  .modal-body {
    height: calc(95vh - #{$HEADER_HEIGHT}px - 15px);
  }
  .sidebar-wrapper {
    height: 100%;
    border: 1px solid $light-steel-blue;
    .sidebar-container {
      float: left;
      width: #{$SIDEBAR_WIDTH}px;
      height: 100%;
      background-color: #F0FFFF;
      border-right: 1px solid #e2e2e2;
    }
    .sidebar-content {
      margin-left: #{$SIDEBAR_WIDTH}px;
      height: 100%;
      background-color: white;
      .tab-content {
        background-color: white;
        height: 100%;
        padding: 17px 20px;
        overflow: scroll;
        .metrics-plot-left {
          float: left;
          width: 250px;
        }
        .metrics-plot-content {
          margin-left: 250px;
        }
      }
    }
  }
}
