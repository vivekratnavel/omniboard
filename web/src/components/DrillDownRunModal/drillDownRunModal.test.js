import React from 'react';
import mockAxios from 'jest-mock-axios';
import {DrillDownRunModal} from './drillDownRunModal';

describe('DrillDownRunModal', () => {
  let wrapper = null;
  const runId = 2;
  const handleCloseMock = jest.fn();
  const status = 'COMPLETED';
  const dbInfo = {key: 'default', name: 'test_db'};
  console.error = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <DrillDownRunModal shouldShow runId={runId} handleClose={handleCloseMock} dbInfo={dbInfo} status={status} localStorageKey='default|view|2'/>
    );
  });

  afterEach(() => {
    // Cleaning up the mess left behind the previous test
    mockAxios.reset();
    jest.clearAllMocks();
    wrapper = null;
  });

  it('should render correctly', async () => {
    expect(wrapper).toMatchSnapshot();
  });
});
