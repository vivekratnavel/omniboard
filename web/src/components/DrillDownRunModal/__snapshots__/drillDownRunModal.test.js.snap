// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrillDownRunModal should render correctly 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="drilldown-run-modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[MockFunction]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Details for Run Id: 2
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <DrillDownView
      dbInfo={
        Object {
          "key": "default",
          "name": "test_db",
        }
      }
      height="100%"
      localStorageKey="default|view|2"
      runId={2}
      stateFilterKeys={
        Array [
          "selectedNavTab",
        ]
      }
      status="COMPLETED"
      width="100%"
    />
  </ModalBody>
</Modal>
`;
